lib/galaxy/tools/actions/metadata.py

```diff
@@ -18,8 +18,11 @@ def execute(self, tool, trans, incoming={}, set_output_hid=False, overwrite=True
         Execute using a web transaction.
         """
         trans.check_user_activation()
-        job, odict = self.execute_via_app(tool, trans.app, trans.get_galaxy_session().id,
-                                          trans.history.id, trans.user, incoming, set_output_hid,
+        session = trans.get_galaxy_session()
+        session_id = session and session.id
```

Comment:
If this is supposed to be a boolean `has_session_id` might be a better variable name, but I think this should probably be `session_id = session.id if session else None` or `session_id = getattr(session, 'id', None)`?

Reply:
I think this already equivalent to `session_id = session.id if session else None`

Reply:
@nsoranzo that is my understanding also, I like @mvdbeek's explicit proposal though

Reply:
Learning something new every day
```python
>>> 'a' and 'b'
'b'
```
I would have expected `True` here.